src/solvers/dgsem_p4est/dg_3d_subcell_limiters.jl

```diff
@@ -0,0 +1,301 @@
+    cache = create_cache(mesh, equations,
+                         VolumeIntegralPureLGLFiniteVolume(volume_integral.volume_flux_fv),
+                         dg, uEltype)
+
+    A4dp1_x = Array{uEltype, 4}
+    A4dp1_y = Array{uEltype, 4}
+    A4dp1_z = Array{uEltype, 4}
+    A4d = Array{uEltype, 4}
+    A5d = Array{uEltype, 5}
+
+    fhat1_L_threaded = A4dp1_x[A4dp1_x(undef, nvariables(equations), nnodes(dg) + 1,
+                                       nnodes(dg), nnodes(dg))
+                               for _ in 1:Threads.nthreads()]
+    fhat1_R_threaded = A4dp1_x[A4dp1_x(undef, nvariables(equations), nnodes(dg) + 1,
+                                       nnodes(dg), nnodes(dg))
+                               for _ in 1:Threads.nthreads()]
+    fhat2_L_threaded = A4dp1_y[A4dp1_y(undef, nvariables(equations), nnodes(dg),
+                                       nnodes(dg) + 1, nnodes(dg))
+                               for _ in 1:Threads.nthreads()]
+    fhat2_R_threaded = A4dp1_y[A4dp1_y(undef, nvariables(equations), nnodes(dg),
+                                       nnodes(dg) + 1, nnodes(dg))
+                               for _ in 1:Threads.nthreads()]
+    fhat3_L_threaded = A4dp1_z[A4dp1_z(undef, nvariables(equations), nnodes(dg),
+                                       nnodes(dg), nnodes(dg) + 1)
+                               for _ in 1:Threads.nthreads()]
+    fhat3_R_threaded = A4dp1_z[A4dp1_z(undef, nvariables(equations), nnodes(dg),
+                                       nnodes(dg), nnodes(dg) + 1)
+                               for _ in 1:Threads.nthreads()]
+    flux_temp_threaded = A4d[A4d(undef, nvariables(equations), nnodes(dg), nnodes(dg),
+                                 nnodes(dg))
+                             for _ in 1:Threads.nthreads()]
+    fhat_temp_threaded = A4d[A4d(undef, nvariables(equations), nnodes(dg),
+                                 nnodes(dg), nnodes(dg)) for _ in 1:Threads.nthreads()]
+    antidiffusive_fluxes = ContainerAntidiffusiveFlux3D{uEltype}(0,
+                                                                 nvariables(equations),
+                                                                 nnodes(dg))
+
+    if have_nonconservative_terms(equations) == true
+        error("Unsupported system of equations with nonconservative terms")
+    end
+
+    return (; cache..., antidiffusive_fluxes,
+            fhat1_L_threaded, fhat1_R_threaded, fhat2_L_threaded, fhat2_R_threaded,
+            fhat3_L_threaded, fhat3_R_threaded, flux_temp_threaded, fhat_temp_threaded)
+end
+
+@inline function subcell_limiting_kernel!(du, u, element,
+                                          mesh::P4estMesh{3},
+                                          nonconservative_terms, equations,
+                                          volume_integral, limiter::SubcellLimiterIDP,
+                                          dg::DGSEM, cache)
+    @unpack inverse_weights = dg.basis
```

Comment:
```suggestion
    @unpack inverse_weights = dg.basis # Plays role of DG subcell sizes
```

I added that comment at the other places